lego-webapp/components/CommandPalette/index.tsx

```diff
@@ -0,0 +1,191 @@
+import { Icon } from '@webkom/lego-bricks';
+import { ArrowDown, ArrowUp, CornerDownRight } from 'lucide-react';
+import { useCallback, useEffect, useState } from 'react';
+import {
+  Autocomplete,
+  Input,
+  Modal,
+  Dialog,
+  DialogTrigger,
+  Menu,
+  MenuItem,
+  MenuSection,
+  Header,
+  ModalOverlay,
+  TextField,
+  Collection,
+  Button,
+  useFilter,
+  Key,
+} from 'react-aria-components';
+import {
+  fetchCommandSuggestions,
+  recordCommandUsage,
+} from '~/redux/actions/UserActions';
+import { useAppDispatch, useAppSelector } from '~/redux/hooks';
+import { selectCommandSuggestions } from '~/redux/slices/users';
+import styles from './CommandPalette.module.css';
+import createCommands from './commands';
+
+const CommandItem = (props: React.ComponentProps<typeof MenuItem>) => (
+  <MenuItem {...props} className={styles.menuItem} />
+);
+
+const CommandPalette = () => {
+  const [isOpen, setOpen] = useState(false);
+  const { contains } = useFilter({ sensitivity: 'base' });
+  const dispatch = useAppDispatch();
+
+  const suggestionIds = useAppSelector(
+    (state) => selectCommandSuggestions(state).slice(0, 3) ?? [],
+  );
+
+  useEffect(() => {
+    dispatch(fetchCommandSuggestions());
+  }, []);
+
+  const commands = createCommands(dispatch, suggestionIds);
+  const allItems = commands.flatMap((s) => s.items);
+
+  const isMac = /mac(os|intosh)/i.test(navigator.userAgent);
+
+  const togglePalette = useCallback(() => setOpen((prev) => !prev), []);
+  const closePalette = useCallback(() => setOpen(false), []);
+
+  const handleCommandSelect = useCallback(
+    (key: Key) => {
+      const id = String(key);
+      const cmd = allItems.find((c) => c.id === id);
+      if (!cmd) return;
+
+      dispatch(recordCommandUsage(cmd.id));
+      cmd.action();
+      closePalette();
+    },
+    [allItems, dispatch, closePalette],
+  );
+
+  useEffect(() => {
+    const handleKeyDown = (e: KeyboardEvent) => {
+      if (e.key === 'k' && (isMac ? e.metaKey : e.ctrlKey)) {
+        e.preventDefault();
+        togglePalette();
+      }
+
+      if (isOpen) {
+        if (e.ctrlKey && e.key.toLowerCase() === 'j') {
+          e.preventDefault();
+          const downEvent = new KeyboardEvent('keydown', {
+            key: 'ArrowDown',
+            code: 'ArrowDown',
+            bubbles: true,
+          });
+          document.activeElement?.dispatchEvent(downEvent);
+          return;
+        }
+        if (e.ctrlKey && e.key.toLowerCase() === 'k') {
+          e.preventDefault();
+          const upEvent = new KeyboardEvent('keydown', {
+            key: 'ArrowUp',
+            code: 'ArrowUp',
+            bubbles: true,
+          });
+          document.activeElement?.dispatchEvent(upEvent);
```

Comment:
For me, opening the modal works with ctrl+k, but the vim bindings do not work. Those bindings open the search bar.

Using Zen on Ubuntu Linux 25.04.

<img width="1173" height="884" alt="Image" src="https://github.com/user-attachments/assets/5089ef99-388d-4d91-bfe7-dc6e93cd0436" />